### DISCUSSION TITLE
Play MPEG-DASH internet radio streams (eg BBC National radio)

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1144] [Bug Reports]
Two separate systems (one an RPi3A with BT speaker and the other an RPi3B with Merus Amp and both with wireless connections) both running update 20221210 plus today's Update branch (for issue #1129) were playing web radio stations and just stopped playing.  Both machines are 'client' machines to another RPI which acts as the (rAudio-1 provided) SharedData server.  The server machine, which has the same level of updates, seems unaffected at the moment.

I thought initially that the web stream had dropped out but it hadn't.  Moving about the UI on both systems was slow to not possible.  I connected, separately, to both systems via SSH and ran journalctl -xe which both showed similar output:

Dec 14 11:58:52 RAConservatory mpd[785]: ffmpeg/http: HTTP error 404 Not Found
Dec 14 11:58:52 RAConservatory mpd[785]: ffmpeg/dash: Failed to open fragment of playlist

over and over again.

I might have expected this immediately after applying the update branch but this was about 30 mins after the update to place.  Both had been playing apparently ok before this event.  I have rebooted both machines to no avail.

@rern - can you shed any light on what might have happened here please ?

**Comment by ceejayemm** [comment]
Further to above I also get:
- message MPD is not running (on player screen).  If I try to start MPD the system hangs
- message Network not connected - but I can still SSH into the machine and ping internal and external sites
- black screens after trying most things, sometimes with a flashing +R logo as it it is trying to do something but what ?  No error messages at this stage

Chris

**Comment by ceejayemm** [comment]
Further info:

 MPD is not running 
mpd 0.23.7-1
# /srv/http/data/mpdconf/mpd.conf
bind_to_address     "0.0.0.0"
db_file             "/srv/http/data/mpd/mpd.db"
music_directory     "/mnt/MPD"
playlist_directory  "/srv/http/data/playlists"
state_file          "/var/lib/mpd/mpdstate"
user                "mpd"
input {
	plugin  "curl"
}
decoder {
	plugin   "ffmpeg"
	enabled  "yes"
}
audio_output {
	name        "SoundCore 2 - A2DP SoundCore 2 - SCO SoundCore 2 | Battery"
	device      "bluealsa"
	type        "alsa"
	mixer_type  "hardware"
}
audio_output {
	name           "On-board - Headphones"
	device         "hw:0,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "none"
}

mpd.service- Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; vendor preset: disabled)
    Drop-In: /usr/lib/systemd/system/mpd.service.d
             └─00-arch.conf
             /etc/systemd/system/mpd.service.d
             └─override.conf
     Active: active (running) since Wed 2022-12-14 16:19:01 GMT; 3min 29s ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
   Main PID: 7107 (mpd)
      Tasks: 7 (limit: 459)
        CPU: 4.421s
     CGroup: /system.slice/mpd.service
             └─7107 /usr/bin/mpd --systemd /srv/http/data/mpdconf/mpd.conf

Dec 14 16:22:29 rAudio-1 mpd[7107]: ffmpeg/http: HTTP error 404 Not Found
Dec 14 16:22:29 rAudio-1 mpd[7107]: ffmpeg/dash: Failed to open fragment of playlist
Dec 14 16:22:29 rAudio-1 mpd[7107]: ffmpeg/http: HTTP error 404 Not Found
Dec 14 16:22:29 rAudio-1 mpd[7107]: ffmpeg/dash: Failed to open fragment of playlist
Dec 14 16:22:29 rAudio-1 mpd[7107]: ffmpeg/http: HTTP error 404 Not Found
Dec 14 16:22:29 rAudio-1 mpd[7107]: ffmpeg/dash: Failed to open fragment of playlist
Dec 14 16:22:30 rAudio-1 mpd[7107]: ffmpeg/http: HTTP error 404 Not Found
Dec 14 16:22:30 rAudio-1 mpd[7107]: ffmpeg/dash: Failed to open fragment of playlist
Dec 14 16:22:30 rAudio-1 mpd[7107]: ffmpeg/http: HTTP error 404 Not Found
Dec 14 16:22:30 rAudio-1 mpd[7107]: ffmpeg/dash: Failed to open fragment of playlist

When I get this message I also get an option to start MPD which after displaying MPD Start... doesn't seem to do anything else

Chris

**Reply by rern** [reply]
Post output of: `amixer -D bluealsa scontrols`

**Reply by ceejayemm** [reply]
@rern 
On the client with a BT speaker this returns nothing:

rAudio-1:~  #  amixer -D bluealsa scontrols
rAudio-1:~  # 

On the client with the Merus Amp (BT not used on this machine) it returns:

RAConservatory:~  #  amixer -D bluealsa scontrols
ALSA lib bluealsa-ctl.c:1126:(_snd_ctl_bluealsa_open) Couldn't get BlueALSA PCM list: The name org.bluealsa was not provided by any .service files
amixer: Mixer attach bluealsa error: No such device

Chris

**Reply by ceejayemm** [reply]
@rern 
Sorry forgot to turn on BT Speaker, so first result which produced nothing, with speaker turned on gives:

rAudio-1:~  #  amixer -D bluealsa scontrols
Simple mixer control 'SoundCore 2 - A2DP',0
Simple mixer control 'SoundCore 2 - SCO',0
Simple mixer control 'SoundCore 2 | Battery',0

Sorry for this confusion.

Chris

**Comment by ceejayemm** [comment]
@rern 
Further info:

Pressing F12 while in the (otherwise blank) Player web page on the affected systems gives:

Uncaught TypeError: Cannot read properties of function.js?v=1671186566
 undefined (reading 'time') 
    at window.onresize (passive.js?v=1671186566:11:4)   
at displayPlayback (function.js?v=1671186566:366:41)

Any help ?

Chris

**Reply by rern** [reply]
Hi Chris,

The SoundCore 2 shows 3 devices which must be handled properly.
Please try the UPDATE branch.

**Comment by ceejayemm** [comment]
@rern 

I could only try to update to the Update branch via SSH as user root, in doing so on machine using the Soundcore2 BT speaker, I get:

rAudio-1:~  #  curl -sfLO https://github.com/rern/rAudio-1/raw/UPDATE/install.sh
rAudio-1:~  #  chmod 755 install.sh
rAudio-1:~  #  ./install.sh "r1 Update UPDATE"
jq: error: Could not open file Update: No such file or directory
jq: error: Could not open file UPDATE.title: No such file or directory

====================================================================================================
 .   {
  "title": "rAudio",
  "version": "20221210",
  "revision": [
    "Fix - Playback - MPD sofeware volume",
    "Fix - Settings - Update status block on change",
    "Fix - Networks - Error when Bluetooth list contains receiver and sender on a single device",
    "Fix - Library - Playlists counts update",
    "Fix - Library - New bookmark missing thumbnail",
    "Fix - Library - Search close button",
    "Fix - Player - ReplayGain setting",
    "Fix - Features - AutoPlay on power on / reboot",
    "Update - Library - Radio Paradise station arts",
    "Improve - Playback - Audio output refresh from no output to connected USB DAC",
    "Improve - Library database update - Set buffer for large database automatically",
    "Improve - Settings - Dropdown menu (switch to Select2)",
    "Change - Run <code>/boot/startup.sh</code> after <code>mpd</code>"
  ],
  "description": "Updates for rAudio",
  "buttonlabel": "Update",
  "nouninstall": 1,
  "thumbnail": "/assets/img/coverart.svg",
  "sourcecode": "https://github.com/rern/rAudio-1",
  "installurl": "https://github.com/rern/rAudio-1/raw/main/install.sh",
  "verify": {
    "command": "[[ $( cat /srv/http/data/addons/r1 ) < 20221007 ]] && echo 1",
    "notverified": "Please download the latest image file to be able to update."
  }
} ...
====================================================================================================
 .  Get files ...
jq: error: Could not open file Update: No such file or directory
jq: error: Could not open file UPDATE.installurl: No such file or directory
curl: option -: is unknown
curl: try 'curl --help' for more information
 !  Get files failed.
rAudio-1:~  #

**Comment by ceejayemm** [comment]
Reflashed a new SD card with i2022117 and then updated to the latest Update branch as of this afternoon (Dec 17th 15:30. Still get same error:

Dec 17 17:09:53 rAudio mpd[1631]: ffmpeg/http: HTTP error 404 Not Found
Dec 17 17:09:53 rAudio mpd[1631]: ffmpeg/dash: Failed to open fragment of playlist

Running journalctl shows the following happening as the Media Player starts:

Dec 17 17:09:52 rAudio systemd[1]: Starting Music Player Daemon...
Dec 17 17:09:52 rAudio mpd[1631]: decoder: Decoder plugin 'wildmidi' is unavailable: configuration file does not exist: /etc/timidity/timi>
Dec 17 17:09:52 rAudio systemd[1]: Started Music Player Daemon.
Dec 17 17:09:52 rAudio audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=mpd comm="systemd" exe="/usr/lib/syste>
Dec 17 17:09:52 rAudio kernel: audit: type=1130 audit(1671296992.819:155): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=mpd comm=">
Dec 17 17:09:52 rAudio audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=mpdidle comm="systemd" exe="/usr/lib/s>
Dec 17 17:09:52 rAudio kernel: audit: type=1130 audit(1671296992.831:156): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=mpdidle co>
Dec 17 17:09:52 rAudio systemd[1]: Started MPD idle.
Dec 17 17:09:53 rAudio mpd[1631]: mixer: Failed to read mixer for 'SoundCore 2 - A2DP': no such mixer control: PCM
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:1309: Getting BlueALSA PCM: PLAYBACK 00:00:00:00:00:00 a2dp
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:1062: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: Setting constraints
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:532: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: Initializing HW
Dec 17 17:09:53 rAudio bluealsa[380]: /usr/bin/bluealsa: D: dbus.c:66: Called: org.bluealsa.PCM1.Open() on /org/bluealsa/hci0/dev_08_EB_ED>
Dec 17 17:09:53 rAudio bluetoothd[285]: /org/bluez/hci0/dev_08_EB_ED_7F_C8_B7/fd0: fd(26) ready
Dec 17 17:09:53 rAudio bluealsa[380]: /usr/bin/bluealsa: D: ba-transport.c:576: New A2DP transport: 18
Dec 17 17:09:53 rAudio bluealsa[380]: /usr/bin/bluealsa: D: ba-transport.c:577: A2DP socket MTU: 18: R:672 W:672
Dec 17 17:09:53 rAudio bluealsa[380]: /usr/bin/bluealsa: D: bluez.c:1247: Signal: org.freedesktop.DBus.Properties.PropertiesChanged(): org>
Dec 17 17:09:53 rAudio bluealsa[380]: /usr/bin/bluealsa: D: ba-transport.c:1174: Starting transport: A2DP Source (SBC)
Dec 17 17:09:53 rAudio bluealsa[380]: /usr/bin/bluealsa: D: ba-transport.c:297: Created BT socket duplicate: [18]: 19
Dec 17 17:09:53 rAudio bluealsa[380]: /usr/bin/bluealsa: D: ba-transport.c:1494: Created new IO thread [ba-a2dp-sbc]: A2DP Source (SBC)
Dec 17 17:09:53 rAudio bluealsa[380]: /usr/bin/bluealsa: D: codec-sbc.c:234: SBC setup: 44100 Hz JointStereo allocation=SNR blocks=16 sub->
Dec 17 17:09:53 rAudio bluealsa[380]: /usr/bin/bluealsa: D: a2dp-sbc.c:223: IO loop: START: a2dp_sbc_enc_thread: A2DP Source (SBC)
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:567: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: FIFO buffer size: 1024 fra>
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:581: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: Selected HW buffer: 4 peri>
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:600: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: Initializing SW
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:600: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: Initializing SW
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:600: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: Initializing SW
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:600: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: Initializing SW
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:639: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: Prepared
Dec 17 17:09:53 rAudio mpd[1631]: D: bluealsa-pcm.c:600: /org/bluealsa/hci0/dev_08_EB_ED_7F_C8_B7/a2dpsrc/sink: Initializing SW
Dec 17 17:09:53 rAudio mpd[1631]: output: Failed to open mixer for 'SoundCore 2 - A2DP': no such mixer control: PCM
Dec 17 17:09:53 rAudio mpdidle.sh[1716]: grep: write error: Broken pipe
Dec 17 17:09:53 rAudio mpd[1631]: ffmpeg/http: HTTP error 404 Not Found
Dec 17 17:09:53 rAudio mpd[1631]: ffmpeg/dash: Failed to open fragment of playlist

So it seems that you are correct about the correct mixer not being chosen but how to change it ?

Chris

**Comment by ceejayemm** [comment]
I have renamed this thread as I now have a better idea as to what is causing this issue.

I have been using the systems below and their configurations for a number of years with few issues. This particular issue started following the application of update 20221110. 

RAOffice
Raspberry PI 3B+ rev 1.2
Original installation version: Unknown
Kernel: 5.15.65-1 aarch64
Version: 20221210
Internet Connection: Wired
SharedData: Server (rAudio-1 provided)
Output Device: Bluetooth, Soundsticks2 Wireless
Music Library: NFS share from OMV 6.x

RAConservatory
Raspberry PI 3B+ rev 1.3
Original installation version: i20221117
Kernel: 5.15.83-2 aarch64
Version: 20221210 plus Update Branch (as at Sat Dec 17) plus pacman -Syu (as at Dec 17)
Internet Connection: WiFi
SharedData: Client (rAudio-1 provided)
Output Device: Merus Amp to QAcoustics i3020 (wired connection)
Music Library: NFS share from OMV 6.x

rAudio-1
Raspberry 3A+ rev 1.0
Original installation version: Unknown
Kernel: 5.15.83-2 aarch64
Version: 20221210 plus Update Branch (as at Sat Dec 17) plus pacman -Syu (as at Dec 17)
Internet Connection: WiFi
SharedData: Server (rAudio-1 provided)
Output Device: Bluetooth, SoundCore2
Music Library: NFS share from OMV 6.x

Being in the UK I tend to listen to BBC radio stations eg

http://as-hls-uk-live.bbcfmt.s.llnwi.net/pool_904/live/uk/bbc_radio_two/bbc_radio_two.isml/bbc_radio_two-audio=320000.norewind.m3u8

but I recently started to hear a message overlaid on their programmes that the stream I was using was due to be removed shortly. The details are shown here:

https://www.bbc.co.uk/sounds/help/questions/recent-changes-to-bbc-sounds/shoutcast-closure

I decided to switch to the suggested replacement MPEG-DASH streams instead, using:

http://a.files.bbci.co.uk/media/live/manifesto/audio/simulcast/dash/uk/dash_full/ak/bbc_radio_two.mpd

This had apparently been working well on all the systems above for two or three weeks prior to the update to 20221210 but after that update I started to get the following error messages:

RAOffice - not affected in any way, no apparent errors

RAConservatory - audio (web radio) output stopped playing and 'journalctl -xe' started to show errors like:
Dec 14 11:58:52 RAConservatory mpd[785]: ffmpeg/http: HTTP error 404 Not Found
Dec 14 11:58:52 RAConservatory mpd[785]: ffmpeg/dash: Failed to open fragment of playlist

over and over again, as shown above. The web pages were unresponsive and although SSH access was occasionally possible, this wasn't always the case.

rAudio-1 - showed similar errors to RAConservatory.

The previous updates to this discussion thread show the investigations done to date. There was some suggestion above that the Bluetooth mixer output was not properly configured however RAConservatory has never used BT output at all.

I concentrated my recent investigations on my rAudio-1 system. I discovered that the MPD/MPC commands from SSH were not being actioned. It was not possible clear the 'playing' playlist, in my case the BBC Radio 2 DASH stream. It would appear that something in the 20221210 update was causing the problems with the DASH based stream. I was able to apply the Update branch as requested above by rern and followed this with a 'pacman -Syu' command which applied many changes. The errors noted above disappeared, the web pages were responsive and there was audio output as normal.

Following this success, I applied the same changes to RAConservatory, however the system was still unresponsive. After many attempts I was able to issue a 'mpd stop' command from SSH, followed by a 'mpc clear' which freed up the web application. 

I temporarily switched back to the Shoutcast based BBC stream on both RAConservatory and rAudio-1 and the errors noted above disappeared and the web pages became responsive again.

It seems that something changed with update 20221210. The DASH streams seem to lock-up both RAConservatory and rAudio-1 however the changes I made above seem to have cured rAudio-1 but not RAConservatory. I don't understand why this should be the case - what is different between the two system, nothing that I can see.  

As a trial, I configured Minimserver to access the BBC DASH streams, which seems to handle them correctly with no errors, and use its streaming capabilities to onwards stream the BBC streams to the rAudio systems.  This is overkill but does in fact provides a temporary solution. This seems to prove that there is a problem with rAudio processing MPEG-DASH streams directly.

@rern - does rAudio currently support the MPEG-DASH streams used by the BBC or can it be made to do so ? I would prefer to access the streams directly in rAudio and remove the temporary Minimserver provided streams from the equation.

An additional side issue is that the output volume levels (on RAConservatory which has wired speakers attached) seem to be lowered with each recent update.  I originally used to run the volume at 17 but subsequent updates have needed this increasing to 28 and latterly to 34 or higher to achieve similar volume output. The 'alsamixer' volume setting DOES change in line with the rAudio volume setting so the old trick of increasing the volume via 'alsamixer' do not seem to apply in this case.

I hope there is a solution to this issue and that above observations will help in some way.

Chris

**Reply by R101G** [reply]
@ceejayemm 
Don't know if it helps, but I think the BBC streams here are the ones to use: 

https://gist.github.com/bpsib/67089b959e4fa898af69fea59ad74bc3

(Edit:  Change it to sbr_high)

**Reply by ceejayemm** [reply]
@R101G - those are (similar to) the ones I originally used for a long time. I think you sent me those originally. They are however due to go out of service next year according to the link in my recent post. 

Thanks

Chris

**Reply by R101G** [reply]
@ceejayemm 
There is a note about legacy links on the page, but nothing about the UK-only feeds. The information there does seem to get updated regularly.
